src_c/music.c

```diff
@@ -51,14 +51,32 @@ mixmusic_callback(void *udata, Uint8 *stream, int len)
 }
 
 static void
-endmusic_callback(void)
+_pg_push_music_event(int type)
 {
-    if (endmusic_event && SDL_WasInit(SDL_INIT_VIDEO)) {
-        SDL_Event e;
-        memset(&e, 0, sizeof(e));
-        e.type = endmusic_event;
-        SDL_PushEvent(&e);
+    pgEventObject *e;
+    SDL_Event event;
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    e = (pgEventObject *)pgEvent_New2(type, NULL);
+    if (e) {
+        pgEvent_FillUserEvent(e, &event);
+#if IS_SDLv1
+        if (SDL_PushEvent(&event) < 0)
+#else
+        if (SDL_PushEvent(&event) <= 0)
```

Comment:
https://wiki.libsdl.org/SDL_PushEvent
> Returns 1 on success, 0 if the event was filtered, or a negative error code on failure; call SDL_GetError() for more information. A common reason for error is the event queue being full. 

I'm wondering about the filtered case.

Reply:
Well, if I get what filtering means, it means that the event passed through the event filter and got rejected right.
In that case we do need to decref the dict

Reply:
Verified, and that’s what filtering means. If the event got filtered then it never entered the event queue.
So we have to check for 0 and decref our dict